socorro/unittest/external/boto/test_connection_context.py

```diff
@@ -218,3 +226,294 @@ def test_fetch_with_regional_s3connection_context(self):
             'us-south-3',
             connection_source
         )
+
+
+
+class MultiplePathsBase(socorro.unittest.testbase.TestCase):
+    """Sets up mocked s3 storage using DatePrefixKeyBuilder"""
+    def setup_mocked_s3_storage(
+        self,
+        executor=TransactionExecutor,
+        executor_for_gets=TransactionExecutor,
+        storage_class='BotoS3CrashStorage',
+        host='',
+        port=0,
+        resource_class=S3ConnectionContext,
+        **extra
+    ):
+        config = DotDict({
+            'resource_class': resource_class,
+            'logger': mock.Mock(),
+            'host': host,
+            'port': port,
+            'access_key': 'this is the access key',
+            'secret_access_key': 'secrets',
+            'bucket_name': 'silliness',
+            'keybuilder_class': DatePrefixKeyBuilder,
```

Comment:
This method is exactly the same as the original one except for this line because we need it to use a `DatePrefixKeyBuilder`. Figured I'd  mention that for easier review.